File: app/src/main/java/com/shariq/vault/ui/screens/MainScreen.kt
package com.shariq.vault.ui.screens

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.spring
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.slideInVertically
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.background
import androidx.compose.foundation.combinedClickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsPressedAsState
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.shariq.vault.model.DocumentType
import com.shariq.vault.model.VaultDocument
import com.shariq.vault.ui.components.GlassmorphicCard
import com.shariq.vault.ui.theme.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MainScreen(
    documents: List<VaultDocument>,
    onAddDocumentClicked: (DocumentType) -> Unit,
    onDocumentClicked: (String) -> Unit,
    onDeleteDocument: (String) -> Unit
) {
    var searchQuery by remember { mutableStateOf("") }
    var selectedCategory by remember { mutableStateOf("All") }
    var showAddMenu by remember { mutableStateOf(false) }
    var documentToDelete by remember { mutableStateOf<VaultDocument?>(null) }

    data class Category(val label: String, val icon: ImageVector)
    val categories = listOf(
        Category("All", Icons.Default.GridView),
        Category("Cards", Icons.Default.CreditCard),
        Category("IDs", Icons.Default.Badge),
        Category("Vehicle", Icons.Default.DirectionsCar)
    )

    val filteredDocuments by remember(searchQuery, selectedCategory, documents.size) {
        derivedStateOf {
            documents.filter { doc ->
                val matchesSearch = searchQuery.isEmpty() ||
                        doc.title.contains(searchQuery, ignoreCase = true) ||
                        (doc.cardNumber?.contains(searchQuery) == true) ||
                        (doc.aadhaarNumber?.contains(searchQuery) == true) ||
                        (doc.panNumber?.contains(searchQuery) == true) ||
                        (doc.dlNumber?.contains(searchQuery) == true) ||
                        (doc.rcNumber?.contains(searchQuery) == true)

                val matchesCategory = when (selectedCategory) {
                    "All" -> true
                    "Cards" -> doc.type == DocumentType.PAYMENT_CARD
                    "IDs" -> doc.type == DocumentType.AADHAAR_CARD ||
                              doc.type == DocumentType.PAN_CARD ||
                              doc.type == DocumentType.DRIVERS_LICENSE
                    "Vehicle" -> doc.type == DocumentType.VEHICLE_RC
                    else -> true
                }
                matchesSearch && matchesCategory
            }
        }
    }

    Scaffold(
        containerColor = CinemaBase,
        floatingActionButton = {
            Box {
                FloatingActionButton(
                    onClick = { showAddMenu = !showAddMenu },
                    containerColor = AccentIndigo,
                    contentColor = Color.White,
                    shape = CircleShape,
                    modifier = Modifier.size(58.dp)
                ) {
                    Icon(
                        imageVector = if (showAddMenu) Icons.Default.Close else Icons.Default.Add,
                        contentDescription = "Add Document",
                        modifier = Modifier.size(26.dp)
                    )
                }

                DropdownMenu(
                    expanded = showAddMenu,
                    onDismissRequest = { showAddMenu = false },
                    modifier = Modifier
                        .background(CinemaElevated)
                        .width(230.dp)
                ) {
                    listOf(
                        Triple(Icons.Default.CreditCard, "Credit / Debit Card", DocumentType.PAYMENT_CARD),
                        Triple(Icons.Default.Badge, "Aadhaar Card", DocumentType.AADHAAR_CARD),
                        Triple(Icons.Default.AssignmentInd, "PAN Card", DocumentType.PAN_CARD),
                        Triple(Icons.Default.DriveEta, "Driver's License", DocumentType.DRIVERS_LICENSE),
                        Triple(Icons.Default.DirectionsCar, "Vehicle RC", DocumentType.VEHICLE_RC)
                    ).forEach { (icon, label, type) ->
                        DropdownMenuItem(
                            leadingIcon = {
                                Icon(icon, contentDescription = null, tint = AccentIndigo, modifier = Modifier.size(18.dp))
                            },
                            text = {
                                Text(label, color = TextPrimary, fontSize = 14.sp, fontWeight = FontWeight.Medium)
                            },
                            onClick = {
                                showAddMenu = false
                                onAddDocumentClicked(type)
                            }
                        )
                    }
                }
            }
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(CinemaBase)
                .padding(paddingValues)
        ) {

            // ── Premium Header ───────────────────────────────────────────
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(
                        brush = Brush.verticalGradient(
                            colors = listOf(CinemaElevated, CinemaBase)
                        )
                    )
                    .padding(horizontal = 24.dp, vertical = 20.dp)
            ) {
                // Subtle ambient blob
                Box(
                    modifier = Modifier
                        .size(120.dp)
                        .align(Alignment.TopEnd)
                        .offset(x = 20.dp, y = (-20).dp)
                        .background(
                            brush = Brush.radialGradient(
                                colors = listOf(AccentGlow, Color.Transparent)
                            ),
                            shape = CircleShape
                        )
                )
                Column {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Box(
                            modifier = Modifier
                                .size(36.dp)
                                .background(AccentGlow, CircleShape),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Lock,
                                contentDescription = "Vault",
                                tint = AccentIndigo,
                                modifier = Modifier.size(18.dp)
                            )
                        }
                        Spacer(modifier = Modifier.width(12.dp))
                        Text(
                            text = "VAULT",
                            fontWeight = FontWeight.ExtraBold,
                            fontSize = 22.sp,
                            letterSpacing = 4.sp,
                            color = TextPrimary
                        )
                    }
                    Spacer(modifier = Modifier.height(6.dp))
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.Shield,
                            contentDescription = null,
                            tint = AccentEmerald,
                            modifier = Modifier.size(12.dp)
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text(
                            text = "${documents.size} encrypted ${if (documents.size == 1) "item" else "items"} · 256-bit AES",
                            color = AccentEmerald,
                            fontSize = 11.sp,
                            fontWeight = FontWeight.SemiBold,
                            letterSpacing = 0.5.sp
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(4.dp))

            Column(modifier = Modifier.padding(horizontal = 20.dp)) {

                // ── Search Bar ───────────────────────────────────────────
                OutlinedTextField(
                    value = searchQuery,
                    onValueChange = { searchQuery = it },
                    placeholder = {
                        Text("Search your vault…", color = TextMuted, fontSize = 14.sp)
                    },
                    leadingIcon = {
                        Icon(
                            Icons.Default.Search,
                            contentDescription = "Search",
                            tint = TextSecondary,
                            modifier = Modifier.size(20.dp)
                        )
                    },
                    trailingIcon = {
                        if (searchQuery.isNotEmpty()) {
                            IconButton(onClick = { searchQuery = "" }, modifier = Modifier.size(32.dp)) {
                                Icon(Icons.Default.Clear, contentDescription = "Clear", tint = TextSecondary, modifier = Modifier.size(16.dp))
                            }
                        }
                    },
                    singleLine = true,
                    shape = RoundedCornerShape(14.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedBorderColor = AccentIndigo,
                        unfocusedBorderColor = CinemaStroke,
                        focusedContainerColor = CinemaSurface,
                        unfocusedContainerColor = CinemaSurface,
                        focusedTextColor = TextPrimary,
                        unfocusedTextColor = TextPrimary,
                        cursorColor = AccentIndigo
                    ),
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 10.dp)
                )

                // ── Category Filter Chips ────────────────────────────────
                LazyRow(
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    contentPadding = PaddingValues(vertical = 4.dp)
                ) {
                    items(categories) { category ->
                        val isSelected = selectedCategory == category.label
                        FilterChip(
                            selected = isSelected,
                            onClick = { selectedCategory = category.label },
                            label = {
                                Text(
                                    category.label,
                                    fontSize = 13.sp,
                                    fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Normal
                                )
                            },
                            leadingIcon = {
                                Icon(category.icon, contentDescription = null, modifier = Modifier.size(15.dp))
                            },
                            colors = FilterChipDefaults.filterChipColors(
                                containerColor = CinemaSurface,
                                selectedContainerColor = AccentIndigo,
                                labelColor = TextSecondary,
                                selectedLabelColor = Color.White,
                                iconColor = TextSecondary,
                                selectedLeadingIconColor = Color.White
                            ),
                            border = FilterChipDefaults.filterChipBorder(
                                enabled = true,
                                selected = isSelected,
                                borderColor = CinemaStroke,
                                selectedBorderColor = AccentIndigo,
                                borderWidth = 0.8.dp,
                                selectedBorderWidth = 0.dp
                            ),
                            shape = RoundedCornerShape(10.dp)
                        )
                    }
                }

                Spacer(modifier = Modifier.height(12.dp))
            }

            // ── Documents List / Empty State ─────────────────────────────
            if (filteredDocuments.isEmpty()) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f),
                    contentAlignment = Alignment.Center
                ) {
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally,
                        modifier = Modifier.padding(40.dp)
                    ) {
                        Box(
                            modifier = Modifier
                                .size(96.dp)
                                .background(
                                    brush = Brush.radialGradient(
                                        colors = listOf(AccentGlow, Color.Transparent)
                                    ),
                                    shape = CircleShape
                                ),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.EnhancedEncryption,
                                contentDescription = null,
                                tint = AccentIndigo.copy(alpha = 0.6f),
                                modifier = Modifier.size(52.dp)
                            )
                        }
                        Spacer(modifier = Modifier.height(20.dp))
                        Text(
                            text = if (searchQuery.isNotEmpty()) "No matching records" else "Vault is empty",
                            color = TextPrimary,
                            fontSize = 19.sp,
                            fontWeight = FontWeight.Bold
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Text(
                            text = if (searchQuery.isNotEmpty())
                                "Try different search keywords."
                            else
                                "Tap + to securely store your Aadhaar, PAN, Driving Licence, RC or payment cards.",
                            color = TextSecondary,
                            fontSize = 13.sp,
                            textAlign = TextAlign.Center,
                            lineHeight = 20.sp
                        )
                    }
                }
            } else {
                LazyColumn(
                    modifier = Modifier.weight(1f),
                    contentPadding = PaddingValues(horizontal = 20.dp, vertical = 0.dp),
                    verticalArrangement = Arrangement.spacedBy(14.dp)
                ) {
                    item { Spacer(modifier = Modifier.height(0.dp)) }
                    items(filteredDocuments, key = { it.id }) { doc ->
                        AnimatedVisibility(
                            visible = true,
                            enter = fadeIn() + slideInVertically(initialOffsetY = { it / 5 })
                        ) {
                            DocumentItem(
                                document = doc,
                                onClick = { onDocumentClicked(doc.id) },
                                onLongClick = { documentToDelete = doc }
                            )
                        }
                    }
                    item { Spacer(modifier = Modifier.height(80.dp)) }
                }
            }
        }
    }

    // ── Delete Confirmation Dialog ───────────────────────────────────────────
    documentToDelete?.let { doc ->
        AlertDialog(
            onDismissRequest = { documentToDelete = null },
            title = { Text("Delete document?", color = TextPrimary, fontWeight = FontWeight.Bold) },
            text = {
                Text(
                    "Permanently delete '${doc.title}'? The encrypted file will be wiped from local storage and cannot be recovered.",
                    color = TextSecondary,
                    lineHeight = 20.sp
                )
            },
            confirmButton = {
                TextButton(onClick = {
                    onDeleteDocument(doc.id)
                    documentToDelete = null
                }) {
                    Text("Delete", color = AccentRed, fontWeight = FontWeight.Bold)
                }
            },
            dismissButton = {
                TextButton(onClick = { documentToDelete = null }) {
                    Text("Cancel", color = TextSecondary)
                }
            },
            containerColor = CinemaElevated,
            shape = RoundedCornerShape(20.dp)
        )
    }
}

// ── Document List Item ───────────────────────────────────────────────────────
@OptIn(ExperimentalFoundationApi::class)
@Composable
fun DocumentItem(
    document: VaultDocument,
    onClick: () -> Unit,
    onLongClick: () -> Unit
) {
    val gradient = when (document.type) {
        DocumentType.PAYMENT_CARD -> CardGradientBluePurple
        DocumentType.AADHAAR_CARD -> CardGradientEmerald
        DocumentType.PAN_CARD -> CardGradientSunset
        DocumentType.DRIVERS_LICENSE -> CardGradientDeepSpace
        DocumentType.VEHICLE_RC -> CardGradientDarkPurple
    }

    val typeIcon = when (document.type) {
        DocumentType.PAYMENT_CARD -> Icons.Default.CreditCard
        DocumentType.AADHAAR_CARD -> Icons.Default.Badge
        DocumentType.PAN_CARD -> Icons.Default.AssignmentInd
        DocumentType.DRIVERS_LICENSE -> Icons.Default.DriveEta
        DocumentType.VEHICLE_RC -> Icons.Default.DirectionsCar
    }

    val typeLabel = when (document.type) {
        DocumentType.PAYMENT_CARD -> document.cardType ?: "Payment Card"
        DocumentType.AADHAAR_CARD -> "Aadhaar Card"
        DocumentType.PAN_CARD -> "PAN Card"
        DocumentType.DRIVERS_LICENSE -> "Driver's Licence"
        DocumentType.VEHICLE_RC -> "Vehicle RC"
    }

    val maskedDetails = when (document.type) {
        DocumentType.PAYMENT_CARD -> formatCardNumber(document.cardNumber)
        DocumentType.AADHAAR_CARD -> formatAadhaarNumber(document.aadhaarNumber)
        DocumentType.PAN_CARD -> formatPanNumber(document.panNumber)
        DocumentType.DRIVERS_LICENSE -> document.dlNumber ?: ""
        DocumentType.VEHICLE_RC -> document.rcNumber ?: ""
    }

    val holderName = when (document.type) {
        DocumentType.PAYMENT_CARD -> document.cardholderName
        DocumentType.AADHAAR_CARD -> document.aadhaarName
        DocumentType.PAN_CARD -> document.panName
        DocumentType.DRIVERS_LICENSE -> document.dlHolderName
        DocumentType.VEHICLE_RC -> document.rcOwnerName
    }

    val validityDate = when (document.type) {
        DocumentType.PAYMENT_CARD -> document.cardExpiry
        DocumentType.DRIVERS_LICENSE -> document.dlExpiry
        DocumentType.VEHICLE_RC -> document.rcExpiry
        else -> null
    }

    val interactionSource = remember { MutableInteractionSource() }
    val isPressed by interactionSource.collectIsPressedAsState()
    val scale by animateFloatAsState(
        targetValue = if (isPressed) 0.97f else 1f,
        animationSpec = spring(dampingRatio = Spring.DampingRatioMediumBouncy, stiffness = Spring.StiffnessMedium),
        label = "cardScale"
    )

    GlassmorphicCard(
        gradient = gradient,
        glowColor = gradient.first().copy(alpha = 0.4f),
        modifier = Modifier
            .scale(scale)
            .combinedClickable(
                interactionSource = interactionSource,
                indication = null,
                onClick = onClick,
                onLongClick = onLongClick
            )
    ) {
        // Header row: type badge + icon
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Top
        ) {
            // Type badge chip
            Box(
                modifier = Modifier
                    .clip(RoundedCornerShape(6.dp))
                    .background(Color.White.copy(alpha = 0.10f))
                    .padding(horizontal = 8.dp, vertical = 3.dp)
            ) {
                Text(
                    text = typeLabel.uppercase(),
                    fontSize = 9.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 1.2.sp,
                    color = Color.White.copy(alpha = 0.85f)
                )
            }
            // Document type icon
            Box(
                modifier = Modifier
                    .size(34.dp)
                    .clip(RoundedCornerShape(8.dp))
                    .background(Color.White.copy(alpha = 0.10f)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = typeIcon,
                    contentDescription = null,
                    tint = Color.White,
                    modifier = Modifier.size(18.dp)
                )
            }
        }

        Spacer(modifier = Modifier.height(14.dp))

        // Title
        Text(
            text = document.title,
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = Color.White,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )

        Spacer(modifier = Modifier.height(16.dp))

        // Masked number
        Text(
            text = maskedDetails,
            fontSize = if (document.type == DocumentType.PAYMENT_CARD) 18.sp else 16.sp,
            fontFamily = FontFamily.Monospace,
            fontWeight = FontWeight.Medium,
            letterSpacing = 2.sp,
            color = Color.White.copy(alpha = 0.90f)
        )

        Spacer(modifier = Modifier.height(18.dp))

        // Footer: holder name + validity
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Bottom
        ) {
            if (!holderName.isNullOrEmpty()) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = "NAME",
                        fontSize = 8.sp,
                        letterSpacing = 1.sp,
                        color = Color.White.copy(alpha = 0.55f)
                    )
                    Text(
                        text = holderName,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color.White,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )
                }
            }
            if (validityDate != null) {
                Column(horizontalAlignment = Alignment.End) {
                    Text(
                        text = if (document.type == DocumentType.PAYMENT_CARD) "EXPIRES" else "VALID UNTIL",
                        fontSize = 8.sp,
                        letterSpacing = 1.sp,
                        color = Color.White.copy(alpha = 0.55f)
                    )
                    Text(
                        text = validityDate,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color.White
                    )
                }
            }
            // Lock watermark at the bottom-right
            if (holderName.isNullOrEmpty() && validityDate == null) {
                Spacer(modifier = Modifier.weight(1f))
            }
            Icon(
                imageVector = Icons.Default.Lock,
                contentDescription = null,
                tint = Color.White.copy(alpha = 0.08f),
                modifier = Modifier
                    .padding(start = 8.dp)
                    .size(28.dp)
            )
        }
    }
}

// ── Number Formatting Helpers ────────────────────────────────────────────────
private fun formatCardNumber(num: String?): String {
    if (num == null) return ""
    val cleaned = num.replace(" ", "")
    if (cleaned.length < 4) return cleaned
    return "•••• •••• •••• " + cleaned.takeLast(4)
}

private fun formatAadhaarNumber(num: String?): String {
    if (num == null) return ""
    val cleaned = num.replace(" ", "")
    if (cleaned.length < 4) return cleaned
    return "•••• •••• " + cleaned.takeLast(4)
}

private fun formatPanNumber(num: String?): String {
    if (num == null) return ""
    if (num.length < 4) return num
    return "••••• " + num.takeLast(4)
}

File: VaultFlutter/lib/ui/screens/main_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:google_fonts/google_fonts.dart';
import '../../models/document.dart';
import '../../providers/vault_provider.dart';
import '../../providers/auth_provider.dart';
import '../theme/colors.dart';
import '../widgets/document_list_item.dart';
import 'add_document_screen.dart';
import 'view_document_screen.dart';

class MainScreen extends ConsumerStatefulWidget {
  const MainScreen({super.key});

  @override
  ConsumerState<MainScreen> createState() => _MainScreenState();
}

class _MainScreenState extends ConsumerState<MainScreen> {
  bool _isReorderMode = false;
  String _searchQuery = '';
  String _activeFilter = 'All';
  final TextEditingController _searchController = TextEditingController();

  final List<Map<String, dynamic>> _categories = [
    {'label': 'All', 'icon': Icons.grid_view},
    {'label': 'Cards', 'icon': Icons.credit_card},
    {'label': 'IDs', 'icon': Icons.badge},
    {'label': 'Vehicle', 'icon': Icons.directions_car},
  ];

  @override
  void dispose() {
    _searchController.dispose();
    super.dispose();
  }

  List<VaultDocument> _applyFilter(List<VaultDocument> docs) {
    List<VaultDocument> filtered;
    switch (_activeFilter) {
      case 'Cards':
        filtered = docs.where((d) => d.type == DocumentType.paymentCard).toList();
      case 'IDs':
        filtered = docs
            .where((d) =>
                d.type == DocumentType.aadhaarCard ||
                d.type == DocumentType.panCard ||
                d.type == DocumentType.genericId)
            .toList();
      case 'Vehicle':
        filtered = docs
            .where((d) =>
                d.type == DocumentType.driversLicense ||
                d.type == DocumentType.vehicleRc)
            .toList();
      default:
        filtered = docs;
    }

    if (_searchQuery.isNotEmpty) {
      final query = _searchQuery.toLowerCase();
      filtered = filtered.where((d) {
        return d.title.toLowerCase().contains(query) ||
            (d.cardNumber?.contains(query) ?? false) ||
            (d.aadhaarNumber?.contains(query) ?? false) ||
            (d.panNumber?.toLowerCase().contains(query) ?? false) ||
            (d.dlNumber?.toLowerCase().contains(query) ?? false) ||
            (d.rcNumber?.toLowerCase().contains(query) ?? false) ||
            (d.genericIdNumber?.toLowerCase().contains(query) ?? false);
      }).toList();
    }
    return filtered;
  }

  void _showAddDocumentMenu() {
    showModalBottomSheet(
      context: context,
      backgroundColor: cinemaElevated,
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.vertical(top: Radius.circular(24)),
      ),
      builder: (ctx) => Padding(
        padding: const EdgeInsets.symmetric(vertical: 24, horizontal: 16),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Padding(
              padding: const EdgeInsets.only(left: 8, bottom: 16),
              child: Text(
                'Add Document',
                style: GoogleFonts.inter(
                  color: textPrimary,
                  fontWeight: FontWeight.w700,
                  fontSize: 18,
                ),
              ),
            ),
            _buildDocTypeOption(ctx, DocumentType.paymentCard, Icons.credit_card, 'Payment Card'),
            _buildDocTypeOption(ctx, DocumentType.aadhaarCard, Icons.badge, 'Aadhaar Card'),
            _buildDocTypeOption(ctx, DocumentType.panCard, Icons.article, 'PAN Card'),
            _buildDocTypeOption(ctx, DocumentType.driversLicense, Icons.drive_eta, "Driver's Licence"),
            _buildDocTypeOption(ctx, DocumentType.vehicleRc, Icons.directions_car, 'Vehicle RC'),
            _buildDocTypeOption(ctx, DocumentType.genericId, Icons.card_membership, 'Generic ID Card'),
          ],
        ),
      ),
    );
  }

  Widget _buildDocTypeOption(
    BuildContext ctx,
    DocumentType type,
    IconData icon,
    String label,
  ) {
    return ListTile(
      leading: Container(
        width: 40,
        height: 40,
        decoration: BoxDecoration(
          color: accentIndigo.withAlpha(26),
          borderRadius: BorderRadius.circular(10),
        ),
        child: Icon(icon, color: accentIndigo, size: 22),
      ),
      title: Text(label, style: GoogleFonts.inter(color: textPrimary, fontSize: 15)),
      onTap: () {
        Navigator.pop(ctx);
        Navigator.push(
          context,
          MaterialPageRoute(
            builder: (_) => AddDocumentScreen(
              documentType: type,
              onSaved: () => Navigator.pop(context),
            ),
          ),
        );
      },
    );
  }

  void _confirmDelete(BuildContext context, VaultDocument doc) {
    showDialog(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('Delete Document'),
        content: Text('Are you sure you want to permanently delete "${doc.title}"?'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx),
            child: const Text('Cancel', style: TextStyle(color: textSecondary)),
          ),
          TextButton(
            onPressed: () async {
              Navigator.pop(ctx);
              await ref.read(vaultProvider.notifier).deleteDocument(doc.id);
            },
            child: const Text('Delete', style: TextStyle(color: accentRed)),
          ),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final vaultState = ref.watch(vaultProvider);

    return Scaffold(
      backgroundColor: cinemaBase,
      body: SafeArea(
        child: Column(
          children: [
            _buildHeader(vaultState),
            _buildSearchBar(),
            _buildFilterRow(),
            Expanded(
              child: vaultState.when(
                loading: () => const Center(
                  child: CircularProgressIndicator(color: accentIndigo),
                ),
                error: (e, _) => Center(
                  child: Text('Error: $e', style: const TextStyle(color: accentRed)),
                ),
                data: (docs) {
                  final filtered = _applyFilter(docs);
                  if (filtered.isEmpty) {
                    return _buildEmptyState(docs.isEmpty);
                  }
                  return ListView.builder(
                    padding: const EdgeInsets.only(bottom: 100),
                    itemCount: filtered.length,
                    itemBuilder: (ctx, idx) {
                      final doc = filtered[idx];
                      return DocumentListItem(
                        document: doc,
                        isReorderMode: _isReorderMode,
                        onTap: _isReorderMode
                            ? null
                            : () {
                                Navigator.push(
                                  context,
                                  MaterialPageRoute(
                                    builder: (_) => ViewDocumentScreen(
                                      documentId: doc.id,
                                      onBack: () => Navigator.pop(context),
                                    ),
                                  ),
                                );
                              },
                        onLongPress: _isReorderMode
                            ? null
                            : () => _confirmDelete(context, doc),
                        onMoveUp: () => ref.read(vaultProvider.notifier).move(doc.id, true),
                        onMoveDown: () => ref.read(vaultProvider.notifier).move(doc.id, false),
                      );
                    },
                  );
                },
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _showAddDocumentMenu,
        tooltip: 'Add Document',
        child: const Icon(Icons.add),
      ),
    );
  }

  Widget _buildHeader(AsyncValue<List<VaultDocument>> vaultState) {
    final count = vaultState.valueOrNull?.length ?? 0;
    return Container(
      decoration: const BoxDecoration(
        gradient: LinearGradient(
          begin: Alignment.topCenter,
          end: Alignment.bottomCenter,
          colors: [cinemaElevated, cinemaBase],
        ),
      ),
      padding: const EdgeInsets.fromLTRB(16, 16, 16, 12),
      child: Column(
        children: [
          Row(
            children: [
              Container(
                width: 40,
                height: 40,
                decoration: BoxDecoration(
                  color: accentIndigo.withAlpha(26),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: const Icon(Icons.lock, color: accentIndigo, size: 20),
              ),
              const SizedBox(width: 12),
              Text(
                'VAULT',
                style: GoogleFonts.inter(
                  color: textPrimary,
                  fontWeight: FontWeight.w800,
                  fontSize: 24,
                  letterSpacing: 4,
                ),
              ),
              const Spacer(),
              IconButton(
                icon: Icon(
                  _isReorderMode ? Icons.check : Icons.reorder,
                  color: _isReorderMode ? accentEmerald : textSecondary,
                ),
                tooltip: _isReorderMode ? 'Done' : 'Reorder',
                onPressed: () => setState(() => _isReorderMode = !_isReorderMode),
              ),
              IconButton(
                icon: const Icon(Icons.lock_outline, color: textSecondary),
                tooltip: 'Lock',
                onPressed: () => ref.read(authProvider.notifier).lock(),
              ),
            ],
          ),
          const SizedBox(height: 8),
          Row(
            children: [
              const Icon(Icons.shield, size: 14, color: accentEmerald),
              const SizedBox(width: 6),
              Text(
                '$count encrypted item${count == 1 ? '' : 's'} · 256-bit AES',
                style: GoogleFonts.inter(
                  color: accentEmerald,
                  fontSize: 12,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildSearchBar() {
    return Padding(
      padding: const EdgeInsets.fromLTRB(16, 8, 16, 4),
      child: TextField(
        controller: _searchController,
        style: GoogleFonts.inter(color: textPrimary, fontSize: 14),
        onChanged: (v) => setState(() => _searchQuery = v),
        decoration: InputDecoration(
          hintText: 'Search documents...',
          prefixIcon: const Icon(Icons.search, color: textSecondary, size: 20),
          suffixIcon: _searchQuery.isNotEmpty
              ? IconButton(
                  icon: const Icon(Icons.clear, color: textSecondary, size: 18),
                  onPressed: () {
                    _searchController.clear();
                    setState(() => _searchQuery = '');
                  },
                )
              : null,
        ),
      ),
    );
  }

  Widget _buildFilterRow() {
    return SizedBox(
      height: 46,
      child: ListView.separated(
        scrollDirection: Axis.horizontal,
        padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 2),
        itemCount: _categories.length,
        separatorBuilder: (_, __) => const SizedBox(width: 8),
        itemBuilder: (ctx, idx) {
          final cat = _categories[idx];
          final String label = cat['label'];
          final IconData icon = cat['icon'];
          final selected = _activeFilter == label;
          return FilterChip(
            showCheckmark: false,
            avatar: Icon(
              icon,
              size: 15,
              color: selected ? Colors.white : textSecondary,
            ),
            label: Text(label),
            selected: selected,
            onSelected: (_) => setState(() => _activeFilter = label),
            backgroundColor: cinemaSurface,
            selectedColor: accentIndigo,
            labelStyle: GoogleFonts.inter(
              color: selected ? Colors.white : textSecondary,
              fontWeight: selected ? FontWeight.bold : FontWeight.normal,
              fontSize: 13,
            ),
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(10),
            ),
            side: BorderSide(
              color: selected ? Colors.transparent : cinemaStroke,
              width: 0.8,
            ),
          );
        },
      ),
    );
  }

  Widget _buildEmptyState(bool noDocsAtAll) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(32),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Container(
              width: 80,
              height: 80,
              decoration: BoxDecoration(
                color: cinemaElevated,
                borderRadius: BorderRadius.circular(20),
              ),
              child: const Icon(
                Icons.enhanced_encryption,
                color: textMuted,
                size: 40,
              ),
            ),
            const SizedBox(height: 20),
            Text(
              noDocsAtAll ? 'Vault is empty' : 'No matching documents',
              style: GoogleFonts.inter(
                color: textPrimary,
                fontWeight: FontWeight.w700,
                fontSize: 18,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              noDocsAtAll
                  ? 'Add your first document using the + button below'
                  : 'Try a different search or filter',
              textAlign: TextAlign.center,
              style: GoogleFonts.inter(color: textSecondary, fontSize: 14, height: 1.6),
            ),
          ],
        ),
      ),
    );
  }
}
